libraries/cli/include/cli/config_updater.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <string>
+
+#include "config/config.hpp"
+
+namespace taraxa::cli {
+
+class ConfigChange {
+ public:
+  ConfigChange() = default;
+  virtual ~ConfigChange() = default;
+
+  virtual bool ShouldApply(uint32_t version) const = 0;
+  virtual void Apply(Json::Value& old_conf, const Json::Value& new_conf) const = 0;
+};
+
+class NetworkIPChange : public ConfigChange {
+ public:
+  NetworkIPChange() = default;
+  virtual ~NetworkIPChange() = default;
+
+  bool ShouldApply(uint32_t version) const;
+  void Apply(Json::Value& old_conf, const Json::Value& new_conf) const;
+};
+
+class ConfigUpdater {
+ public:
+  ConfigUpdater(int network_id);
+  ~ConfigUpdater() = default;
+
+  void UpdateConfig(const std::string& config_name, Json::Value& old_conf);
+
+ private:
+  Json::Value new_conf_;
+  std::vector<std::unique_ptr<ConfigChange>> config_changes_;
```

Comment:
Do we need this while boilerplate around ConfigChange class and derived classes.... ? Wouldnt be somehting like this  simpler and achieve the same thing ?

struct ConfigChange {
  VersionType version;
  std::function<void(const std::string& config_name, Json::Value& old_conf)> apply;
}

std::vector<ConfigChange> config_changes_;

private:
void registerConfigChanges() {
  // register all config changes functions

  // IP config change
  ConfigChange networkIPChange(1.0.0, [](Json::Value& old_conf, const Json::Value& new_conf){ code... });
 config_changes_.push_back(networkIPChange)
}